pyuvsim/source.py

```diff
@@ -89,54 +133,54 @@ def coherency_calc(self, time, telescope_location):
         but must be rotated into local alt/az.
 
         Args:
-            time: astropy Time object
             telescope_location: astropy EarthLocation object
 
         Returns:
             local coherency in alt/az basis
         """
-        if not isinstance(time, Time):
-            raise ValueError('time must be an astropy Time object. '
-                             'value was: {t}'.format(t=time))
-
         if not isinstance(telescope_location, EarthLocation):
             raise ValueError('telescope_location must be an astropy EarthLocation object. '
                              'value was: {al}'.format(al=telescope_location))
+        # stokes shape = (4, Ncomponents)
+        # ra/dec/hour_angle shape = (1,Ncomponents)
+        # Ionly_mask = (Ncomponents)
+        Ionly_mask = np.sum(self.stokes[1:, :], axis=0) == 0.0
```

Comment:
The implementation here really confuses me and it's going to be very hard to rebase the work James has been doing on it. Could we instead split the array into polarized and unpolarized and then only call the polarized specific code if there are any polarized sources?

Reply:
This is doing the rotation matrix calculation for all sources and then overwriting the rotations as identity matrices for sources with only stokes-I. For now, it probably makes more sense to do the opposite order. The only change is that the rotation matrices now have an `Ncomponents` axis as the last axis.

Looking at the `coherency_basis_rotation_redux` branch, it seems like it will take some effort to vectorize all of the various functions involved.

Reply:
Ok. I think you will almost always have many more unpolarized sources, so doing the hard work for only the polarized sources make sense. I agree there will be some work to vectorize the polarized stuff, but keeping the unpolarized ones separate will help, I think. We might want to consider doing a squash on the commits on this branch to help with the rebasing.